src/run/run_environment/local/provider.rs

```diff
@@ -31,78 +33,68 @@ impl TryFrom<&Config> for LocalProvider {
     type Error = Error;
     fn try_from(config: &Config) -> Result<Self> {
         let current_dir = std::env::current_dir()?;
-        match find_repository_root(&current_dir) {
+
+        if let Some(RepositoryOverride {
```

Comment:
Seen together, revert this logic to the previous behavior.
Keep the fake commit with the correct length though

Reply:
I made the logic a bit clearer by keeping the early return when not in a git repository, but reverted back